app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -176,4 +178,32 @@ def parse_image(image, is_public)
 
     return uid, new_result
   end
+
+  def parse_vapp_template(vapp_template)
+    uid = vapp_template.id
+
+    begin
+      # The content of the template is the OVF specification of the vApp template
+      content = @connection.get_vapp_template_ovf_descriptor(uid).body
+
+      new_result = {
+        :type        => ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate.name,
+        :ems_ref     => uid,
+        :name        => vapp_template.name,
+        :description => vapp_template.description,
+        :orderable   => false,
+        :content     => content,
+        # By default #save_orchestration_templates_inventory does not set the EMS
+        # ID because templates are not EMS specific. We are setting the EMS
+        # explicitly here, because vapps are specific to concrete EMS.
+        :ems_id      => @ems.id
+      }
+
+      return uid, new_result
+    rescue
+      # Without content we should not create the orchestration template because
+      # these are identified by it (or its MD5)
+      $log.info("MIQ(#{self.class.name}.#{__method__}) Downloading vApp template for vApp template #{uid} failed")
```

Comment:
So the problem is if we skip the template it will be deleted from the database, so unless getting the content is a much more likely call to fail on network issues it'd almost be better to fail the whole inventory refresh since we couldn't get all the information we need, thoughts?